### ISSUE TITLE
Catch detailed error messages

### ISSUE THREAD

**Issue by bence-b** [O365/python-o365#754]
Hi
Thanks for this cool module it really makes my life easier!
Is there a way to capture the details on the Request Errors?
For example, if I'm trying to refresh the token and an error occurs all I got is a "RuntimeError: Token Refresh Operation not working". My goal is to know whether it was a ConnectionError or an invalid request etc. 
Because the exceptions are captured and logged to the stdout, I'm unable to capture them.

Thanks,
Ben

**Comment by janscas** [comment]
Yes, just set the logging to output the requests error messages.
